src/main/groovy/jenkins/automation/builders/MultibranchPipelineJobBuilder.groovy

```diff
@@ -0,0 +1,63 @@
+package jenkins.automation.builders
+import jenkins.automation.utils.CommonUtils
+
+import javaposse.jobdsl.dsl.DslFactory
+import javaposse.jobdsl.dsl.jobs.MultibranchWorkflowJob
+
+
+/**
+ * Multibranch Pipeline Job builder creates a set of Pipeline projects according to detected branches in one SCM repository
+ * @param name Sets the job name
+ * @param description Sets a description for the folder
+ * @param branchSourceGithub if true, provide  ghOrganizationName, ghRepositoryName, ghScanCredentialsId and ghEndpoint
+ * @param ghOrganizationName Sets the name of the GitHub Organization or GitHub User Account
+ * @param ghRepositoryName Sets the name of the GitHub repository
+ * @param ghScanCredentialsId Sets scan credentials for authentication with GitHub
+ * @param ghEndpoint Sets the GitHub API URI
+ * @param branchSourceGit if true, provide ghRemote and ghCredentialsId
+ * @param ghRemote Sets the Git remote repository URL
+ * @param ghCredentialsId Sets credentials for authentication with the remote repository 
+ */
+class MultibranchPipelineJobBuilder {
+    String name
+    String description
+    Boolean branchSourceGithub
+    String ghOrganizationName
+    String ghRepositoryName
+    String ghScanCredentialsId
+    String ghEndpoint
+    Boolean branchSourceGit
+    String ghRemote
+    String ghCredentialsId
+    Boolean discardOldItems
+    int oldNumToKeep
+
+    MultibranchWorkflowJob build(DslFactory factory) {
```

Comment:
This should be multi branch pipeline  job: https://jenkinsci.github.io/job-dsl-plugin/#method/javaposse.jobdsl.dsl.DslFactory.multibranchPipelineJob

MultiBranchWorkflowJob has been deprecated in 1.47: https://jenkinsci.github.io/job-dsl-plugin/#path/multibranchWorkflowJob

Reply:
@imuchnik the factory still returns a class with the old name, no? https://github.com/jenkinsci/job-dsl-plugin/blob/419c637c11aa7fc284e73536097df57bdce4fc29/job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/DslFactory.groovy#L184

Reply:
@acrewdson Yes, for now, but it is marked as deprecated and will go away in couple of releases. See this for reference: https://issues.jenkins-ci.org/browse/JENKINS-33325

Reply:
@imuchnik I know it's deprecated. What class should we be saying `build` returns, then?

Reply:
@acrewdson  multibranchPipelineJob

Reply:
@imuchnik a class name that starts with a lowercase letter? Where are you seeing that in the `job-dsl` codebase?

Reply:
@acrewdson  you are right. I assumed that multibranch job inherits from Job, but in fact, it inherits from folder. This should work.